utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,551 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+# This program is run by "Model Auto-tracing & Uploading" workflow
+# (See model_uploader.yml) to perform model auto-tracing and prepare
+# files for uploading to OpenSearch model hub.
+
+import argparse
+import os
+import shutil
+import sys
+import warnings
+from typing import List, Optional, Tuple
+from zipfile import ZipFile
+
+import numpy as np
+from mdutils.fileutils import MarkDownFile
+from numpy.typing import DTypeLike
+from sentence_transformers import SentenceTransformer
+
+THIS_DIR = os.path.dirname(__file__)
+ROOT_DIR = os.path.join(THIS_DIR, "../..")
+sys.path.append(ROOT_DIR)  # Required for importing OPENSEARCH_TEST_CLIENT
+
+LICENSE_PATH = "LICENSE"
+from opensearch_py_ml.ml_commons import MLCommonClient
+from opensearch_py_ml.ml_models.sentencetransformermodel import SentenceTransformerModel
+from tests import OPENSEARCH_TEST_CLIENT
+
+BOTH_FORMAT = "BOTH"
+TORCH_SCRIPT_FORMAT = "TORCH_SCRIPT"
+ONNX_FORMAT = "ONNX"
+
+TEMP_MODEL_PATH = "temp_model_path"
+ORIGINAL_FOLDER_PATH = "sentence-transformers-original/"
+TORCHSCRIPT_FOLDER_PATH = "sentence-transformers-torchscript/"
+ONNX_FOLDER_PATH = "sentence-transformers-onnx/"
+UPLOAD_FOLDER_PATH = "upload/"
+MODEL_CONFIG_FILE_NAME = "ml-commons_model_config.json"
+OUTPUT_DIR = "trace_output/"
+LICENSE_VAR_FILE = "apache_verified.txt"
+TEST_SENTENCES = [
+    "First test sentence",
+    "This is a very long sentence used for testing model embedding outputs.",
+]
+RTOL_TEST = 1e-03
+ATOL_TEST = 1e-05
+ML_BASE_URI = "/_plugins/_ml"
+
+
+def verify_license_in_md_file() -> bool:
```

Comment:
```
from transformers import AutoModel, AutoConfig

def check_model_license(model_name_or_path):
    config = AutoConfig.from_pretrained(model_name_or_path)
    license_text = config.license
    return "apache-2.0" in license_text.lower()

# Example usage:
model_name_or_path = "bert-base-uncased"
is_apache_2_0 = check_model_license(model_name_or_path)

if is_apache_2_0:
    print(f"The model '{model_name_or_path}' is under the Apache 2.0 license.")
else:
    print(f"The model '{model_name_or_path}' is not under the Apache 2.0 license.")
```

Can you please give a try and check if that works?

Reply:
Okay. Will try this.

Reply:
`AutoConfig` has no `license` attribute. I looked at other Huggingface object types and haven't found any license attribute as well. Based on https://huggingface.co/docs/hub/repositories-licenses, it is probably only specified in metadata of readme file.
<img width="356" alt="Screenshot 2023-08-07 at 11 33 44 PM" src="https://github.com/opensearch-project/opensearch-py-ml/assets/106889996/f4da37c7-83c1-4832-8cc5-594ceae86bb5">
ute as well. Based on